### DISCUSSION TITLE
How to single out the Stanford Core NLP Models Spanish file?

### DISCUSSION THREAD

**Discussion by edump72** [stanfordnlp/CoreNLP#1407] [Q&A] [answered]
I am using Android Studio. I have the Stanford Core NLP Models 4.2.0 jar file, but I want to massively reduce the space of the jar file so I want to eliminate every other redundant file :

![Captura de pantalla (717)](https://github.com/stanfordnlp/CoreNLP/assets/100035955/6ab4aec5-349e-4e79-b722-2cc2f1e11bd6)

This is the current loading of the model, and all works perfectly, but as I have said, it is massively heavy and slow :

![Captura de pantalla (719)](https://github.com/stanfordnlp/CoreNLP/assets/100035955/d4931fa9-b061-4117-b7b9-354bdb0bb002)

I have tried to add on the libs folder only the spanish model I want and also to name all the paths involving it, but the result has been nullPointerExceptions as according to Android Studio, the file does not exist :

![Captura de pantalla (720)](https://github.com/stanfordnlp/CoreNLP/assets/100035955/6a9237bc-823e-428c-8459-2b173649ae01)

I will be pleased if somebody could help me . Thanks in advance!

**Comment by AngledLuffa** [comment]
There's a "jar" tool that ships with JDK that can be used to take apart and
rebuild the jar file to meet your exact specifications

**Reply by edump72** [reply]
Thank you so much for responding, I haven’t found help anywhere else. Would it be too much trouble to tell me where that tool is?

**Reply by AngledLuffa** [reply]
How can I say "no" when you ask so nicely?

I have java installed on Centos Stream using some package manager, and the "jar" tool went in the same place as java itself:

```
[john@localhost ~]$ which java
/usr/bin/java
[john@localhost ~]$ which jar
/usr/bin/jar
```

whereas on Windows, I installed using the java installer, and it also went to the same place as the java executable:

```
c:\Users\horat\nlp\codebase\CoreNLP>which java
which java
/c/Program Files/Java/jdk-14.0.2/bin/java

c:\Users\horat\nlp\codebase\CoreNLP>which jar
which jar
/c/Program Files/Java/jdk-14.0.2/bin/jar
```

and all I need to do to get a detailed list of options is run `jar --help`, with `-c` and `-x` being the most relevant ones for your use case:

```
c:\Users\horat\nlp\codebase\CoreNLP>jar --help
jar --help
Usage: jar [OPTION...] [ [--release VERSION] [-C dir] files] ...
jar creates an archive for classes and resources, and can manipulate or
restore individual classes or resources from an archive.

  -c, --create               Create the archive
  -x, --extract              Extract named (or all) files from the archive
```

**Reply by edump72** [reply]
Ok seriously, thank you so much again. I now have been able to extract only the necessary model for my app to work, however at the time of naming the model to access to it I always get an error as mentioned in the first post. The name of the file is .gz. Do you have any idea of how to possibly name it without getting a NullPointerException?
![Captura de pantalla (720)](https://github.com/stanfordnlp/CoreNLP/assets/100035955/ab0c40e3-2b44-4180-abda-8a94b5950aa7)

**Comment by AngledLuffa** [comment, marked as answer]
I would say you need to provide the complete path (edu/stanford/...)

**Reply by edump72** [reply]
Ok.Thanks so much!
